src/crypto/crypto_common.cc

```diff
@@ -508,6 +508,54 @@ MaybeLocal<Value> GetRawDERCertificate(Environment* env, X509* cert) {
   return Buffer::New(env, ab, 0, ab->ByteLength()).FromMaybe(Local<Object>());
 }
 
+MaybeLocal<Value> GetExtensions(Environment* env, X509* cert) {
+  Local<Object> extensions = Object::New(env->isolate());
+
+  const STACK_OF(X509_EXTENSION)* ext_list = X509_get0_extensions(cert);
+  int num_extensions = sk_X509_EXTENSION_num(ext_list);
+
+  for (int i = 0; i < num_extensions; ++i) {
+    X509_EXTENSION* ext = sk_X509_EXTENSION_value(ext_list, i);
+    const char* ext_name =
+        OBJ_nid2sn(OBJ_obj2nid(X509_EXTENSION_get_object(ext)));
+
+    BIO* ext_bio = BIO_new(BIO_s_mem());
+    if (!ext_bio) {
+      continue;
+    }
+
+    int result = X509V3_EXT_print(ext_bio, ext, 0, 0);
+    if (result <= 0) {
+      BIO_free(ext_bio);
+      continue;
+    }
+
+    char* ext_value_buf;
+    int64_t ext_value_len = BIO_get_mem_data(ext_bio, &ext_value_buf);
+
+    if (!ext_value_buf || ext_value_len <= 0) {
+      BIO_free(ext_bio);
+      continue;
+    }
+
+    Local<String> ext_value_str = String::NewFromUtf8(env->isolate(),
+                                                      ext_value_buf,
+                                                      NewStringType::kNormal,
+                                                      ext_value_len)
+                                      .ToLocalChecked();
+
+    BIO_free(ext_bio);
+
+    extensions
+        ->Set(env->context(),
+              String::NewFromUtf8(env->isolate(), ext_name).ToLocalChecked(),
+              ext_value_str)
+        .IsNothing();
```

Comment:
It makes no sense to call `IsNothing()` without checking the return value. You should handle this error by propagating it to the JavaScript caller.